src/filters/qualitative-filter.js

```diff
@@ -0,0 +1,221 @@
+    })
+  }
+
+  const handleListItemOnClick = (e, opt) => {
+    e.stopPropagation()
+
+    let arr = (optionsValue || '').split(',').filter((v) => v)
+    if (!arr.length) {
+      arr = [...options]
+    }
+
+    const i = arr.indexOf(opt)
+    if (i > -1) {
+      arr.splice(i, 1)
+    } else {
+      arr.push(opt)
+    }
+
+    if (!optionsValue || optionsValue.includes(opt)) {
+      onlyButtonContainer[options.indexOf(opt)].style.display = 'none'
+    } else {
+      onlyButtonContainer[options.indexOf(opt)].style.display = 'initial'
+    }
+
+    setOptionsValue(arr.length && arr.length < options.length ? arr.join(',') : undefined)
+  }
+
+  const handleOnlyOnClick = (e, opt) => {
+    e.stopPropagation()
+    setOptionsValue(opt)
+  }
+
+  const applyOnClick = (e) => {
+    e.stopPropagation()
+    closePopper && closePopper(e)
+    setFilter(optionsValue)
+  }
+
+  const cancelOnClick = (e) => {
+    e.stopPropagation()
+    closePopper && closePopper(e)
+    setOptionsValue(filterValue)
+  }
+
+  return (
+    <div>
+      <List ref={listItemRef} className={classes.list}>
+        <TextField fullWidth variant='outlined'
+          size='small'
+          InputProps={{
+            endAdornment: (
+              <InputAdornment position='end'>
+                <SearchIcon />
+              </InputAdornment>
+            ),
+            'aria-label': 'search',
+          }}
+          onClick={(e) => { e.stopPropagation() }}
+          onChange={handleOnSearch}
+          onKeyUp={filterList}
+          value={value || ''}
+          placeholder={`Search in ${options.length} records...`}
+        />
+        {options.map((opt, index) => {
+          const labelID = `${id}-selection-label-${opt}`
+          return (
+            <div key={opt} className={classes.listItemContainer} onMouseEnter={() => handleOnMouseEnter(opt, index)} onMouseLeave={() => handleOnMouseLeave(index)}>
+              <ListItem
+                className={classes.MuiListItem}
+                role={undefined}
+                dense
+                button
+                onClick={(e) => {
+                  handleListItemOnClick(e, opt)
+                }}
+              >
+                <ListItemIcon className={classes.MuiListItemIcon}>
+                  <Checkbox
+                    color='primary'
+                    edge='start'
+                    checked={!optionsValue || (optionsValue || '').includes(opt)}
+                    tabIndex={-1}
+                    disableRipple
+                    inputProps={{ 'aria-labelledby': labelID }}
+                  />
+                </ListItemIcon>
+                <ListItemText className='listItemText' id={labelID} primary={opt} />
+              </ListItem>
+
+              <Button
+                className={classes.onlyButton}
+                color="primary"
+                onClick={(e) => {
+                  handleOnlyOnClick(e, opt)
+                }}
+              >
+                Only
```

Comment:
If it is possible, it'd be nice to add mutually exclusive filtering option to the button which deselect all options except for the one selected like the example below:

1. Click on `Only` to deselect all items except for `Ontario`
![Screen Shot 2021-04-20 at 5 13 34 PM](https://user-images.githubusercontent.com/59836949/115464967-dbb70500-a1fb-11eb-870b-1f1c7bbb92e7.png)

2. Filter shows the except button when only `Ontario` is selected.
![Screen Shot 2021-04-20 at 5 13 10 PM](https://user-images.githubusercontent.com/59836949/115464975-dfe32280-a1fb-11eb-8e5a-16dfbcabbf31.png)

3. Click on `Except` deselects `Ontario` and select rest of the items. 
![Screen Shot 2021-04-20 at 5 14 00 PM](https://user-images.githubusercontent.com/59836949/115464987-e2457c80-a1fb-11eb-8191-f51d4bd7bb2e.png)